### DISCUSSION TITLE
Expanding the Membership of FIP Editors

### DISCUSSION THREAD

**Discussion by luckyparadise** [filecoin-project/FIPs#783] [Enhancements - Governance]
I'd like to start an open discussion about the current state of our FIP Editorial process. With the growing ecosystem and increasing activity in our repositories, we've observed significant delays in the editorial and technical review processes for FIPs and PRs. Our current team of FIP Editors, consisting of @anorth, @momack, @arajasek, @jennijuju, @kaitlin-beegle, and @raulk, has been doing an outstanding job. However, in order to meet the heightened demand and ensure timely reviews, it is paramount that we expand this team. As our community and the number of contributions grow, it's essential that our editorial process scales accordingly.

## Problem Statement:

- **Delays**: Lengthy review times for PRs and FIPs.
- **Bottlenecks**: Current number of FIP Editors might not be sufficient to handle the volume of FIPs submitted.
- **Efficiency**: Faster editorial checks will lead to swifter progress in protocol upgrades and features.

## Proposed Solution:

Expand the current membership of FIP Editors. This would mean bringing more qualified individuals into the fold who can carry out both editorial and technical checks. By doing so, we can distribute the workload, reduce wait times, and ensure a faster turnaround on FIPs and PRs.

## Recommendation:

I propose that we open up nominations for new FIP Editors. The nomination process would allow community members or Core Developers to suggest individuals they believe would be apt for the role. Each nominee should ideally have a proven track record in the Filecoin community or possess the necessary expertise to be an effective editor.

-----

Kindly provide your thoughts, feedback, and nominations. I am especially interested in hearing from the core devs and current FIP Editors on this matter.

Thank you!

**Comment by kaitlin-beegle** [comment]
Hi @luckyparadise, thanks for raising this. 

In general, I think we should always be open to adding or removing FIP Editors.  Like many other things, there are no strict provisions in FIP0001 that guide us on how exactly to manage this process.  So, I appreciate you flagging the need. 

Though I don't think the current number and cadence of FIP Editor work is necessarily preventing FIPs from getting where they need to go, I would like to see an overall increase in the velocity of FIP Editor reviews. IMO, FIP Editors should be the first ones reviewing and editing drafts, and should be pulling in others whose expertise is useful in auditing and peer reviewing specs and their proposed impacts.  Again, I think current FIP Editors are good at doing this, but everyone is busy; it's easy for this responsibility to get put on the back burner (speaking from experience *cough* sorry *cough*) and for a few rounds of reviews to take months. Again, not necessarily a bad thing, but certainly worth thinking about improving.  We could always use more help. 

My only concern is that the role of FIP Editor is difficult.  It's sort of a voluntary role, with a high level of authorization to review and merge drafts.  Ultimately, who gets to decide who gets to be a FIP Editor? To vet their qualifications, and approve them for increased permissions over the repo? **Should someone wish to be a FIP Editor, I think all current FIP Editors should be unanimous in agreeing to having them join.**

**Reply by jennijuju** [reply]
> FIP Editors should be the first ones reviewing and editing drafts, and should be pulling in others whose expertise is useful in auditing and peer reviewing specs and their proposed impacts.

I would argue this is a responsibility for FIP authors too, its similar to engineers are responsible for getting their code (a.k.a their work) reviewed, iterated and shipped (or closed).

**Comment by Fatman13** [comment]
Great proposal! 👍 I think it's great to expand FIP editors so that we have a more diverse group of people to shed lights on subject matters from their perspectives. Right now the board of FIP editors are composed of PL / FF employees only, which is a bit of monotonous for a community-driven decentralized project to say the least. 

> I propose that we open up nominations for new FIP Editors.

In light of this, I nominate myself to be a candidate for the expansion of FIP editors. I hope that with my contribution to the FIP governance process as an FIP editor, we can build a better tomorrow for the Filecoin network, a prosperous future in a jungle of highly competitive public chains. With regard to my resume, I have been a Filecoin community member since Space Race and have a track record of active participation in FIP governance. Noticeably that I was the co-author of [FIP0014](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0014.md) along with many comments and inputs in different threads of FIP [discussions](https://github.com/filecoin-project/FIPs/discussions/554#discussioncomment-5389118). What's more, I specialize in bringing in Chinese speaking community members' perspectives into the FIP decision making process. Looking forward to have your support on my candidacy!

**Comment by anorth** [comment]
I concur we should expand the group. Kaitlin's proposal of unanimous assent sounds like a reasonable way to do this while the group is still small, so long as we actually find candidates we can assent to. If we can't, we'll need to relax that bar.

**Comment by jennijuju** [comment]
> I think all current FIP Editors should be unanimous in agreeing to having them join.

I think this makes sense, just so that we can help make sure that the potential candidate understands the responsibility & guidelines & flow of being an FIP editor.


@filecoin-project/fips-editors That being said, I would like to officially propose @jsoares to join FIP editors. He has been playing an important role in Filecoin development -  he actively contributed to Drand (a critical external service Filecoin is dependent on) and is leading the effort on Filecoin scalability and consensus research and development at [consensus lab](https://consensuslab.world).

He has also been helping reviewing FIPs and give good suggestion on FIPs process from **editorial perspective** (also often with refreshing insights), see examples: [[1](https://github.com/filecoin-project/FIPs/pull/1)], [[2](https://github.com/filecoin-project/FIPs/pull/552)], [[3](https://github.com/filecoin-project/FIPs/pull/788#issuecomment-1684946688)].  He has also experiences with providing governance within other communities, for example as mentioned [here](https://research.protocol.ai/authors/jorge-soares/),"He’s a Senior Member of [IEEE](https://www.ieee.org/) and a long-time volunteer, currently serving as Vice-Chair of the European Public Policy Committee."

@jsoares has kindly agreed to help us out if existing FIP editors 👍 , and I think he will be a great asset to the effort!

**Reply by anorth** [reply]
+1 for @jsoares

**Reply by raulk** [reply]
+1. The times I've seen @jsoares chime in concerning process, he's voiced well-reasoned, sensible arguments, often drawing insightful analogies with other existing or historical systems. I expect having him as an editor will level up our collective game.

**Reply by kaitlin-beegle** [reply]
Also +1 on @jsoares. Current editors are over capacity in their work, and expanding the pool with qualified people would be really helpful.

**Comment by luckyparadise** [comment]
I do agree with all your views regarding this. I think we should not aim to drastically open up and expand the group in a short period of time. 

Moving ahead, I think there is a general agreement in terms of the way forward for this process. vv

Self-nominate/Nominate (with rationale for nomination)  - Nomination is presented to current Editors - Decision is made regarding the nomination.
